package.json

```diff
@@ -86,20 +98,29 @@
     "rimraf": "2.5.4",
     "sass": "^1.37.4",
     "semantic-release": "^17.0.4",
-    "sinon": "^1.17.4"
+    "sinon": "^1.17.4",
+    "tsd": "^0.17.0",
+    "typescript": "^4.3.5"
   },
   "scripts": {
     "prebuild": "yarn run clean",
-    "build": "BABEL_ENV=production babel src -d dist && cp -r ./src/*.png ./dist && sass ./src/intlTelInput.scss ./dist/main.css",
+    "build": "yarn compile:js && yarn compile:dts && yarn compile:css && yarn compile:png",
+    "compile:js": "BABEL_ENV=production babel src -d dist",
+    "compile:dts": "rsync -avh --include='*/' --include='*.d.ts' --exclude='*' src/ dist --prune-empty-dirs",
+    "compile:css": "sass ./src/intlTelInput.scss ./dist/main.css",
+    "compile:png": "cp -r -v ./src/*.png ./dist",
```

Comment:
In my previous attempt https://github.com/patw0929/react-intl-tel-input/pull/351#discussion_r442344390 I split this up for readability. The only difference is that instead of [`gulp`](https://github.com/patw0929/react-intl-tel-input/pull/351#discussion_r442344390), I chose to stick with the macOS's `rsync` ability, which means less dependency bloat.

Reply:
The one downside of relying `rsync` is that we make assumptions about the host system. For instance, my system doesn't have `rsync` out of the box. :sweat_smile:

In this case, `rsync` is used to grab all the definition files and copy them over to `dist`. An alternative soln to this could be to have a `typing` folder at the root (i.e. `src/typing`) which would contain the type definition files, which we could then copy easily in one go. From there, we probably would need some import/export tweaks, but would otherwise be in an equivalent state.

Another alternative (which might admittedly be a bit more janky) would be to just explicitly `cp src/components/*.d.ts dist/components/*.d.ts` and remove the dependency on `rsync`.

Any thoughts?

Reply:
@mcataford Sorry, I'm late! Busy week.

I found this alternative a couple of days ago, hoping that it's a little more graceful than `cp`ing the folders one by one explicitly:

```sh
cd src && find * -type f -name "*.d.ts" -print -exec cp -r . ../dist \; ; cd ..
```

I've heard that [`find *`](https://stackoverflow.com/a/2596474) isn't going to evaluate files and folders starting with `.` (often used for hidden stuff) but I don't foresee use case for us, e.g. `src/.components`, `src/.under.the.hood.js`, so I figured that this script is still workable. I'm partial towards preserving the file structure as is because it's easy to spot which `.js` files don't have a matching `.d.ts` to go with it.

What do you think of it?

Reply:
That looks good to me! Thanks for addressing! :rocket:

Reply:
Nice!